src/app/pages/data-view-export/view-records/view-records.component.ts

```diff
@@ -186,30 +195,79 @@ export class ViewRecordsComponent implements OnInit {
     }
 
     public export() {
-        this.apiService.exportRecords(this.dateStart, this.dateEnd, this.speciesName, this.selectedDataset.id,
-            this.fileType).subscribe(
-            resp => {
-                const timeStamp = moment().format('YYYY-MM-DD-HHmmss');
-                const extension = this.fileType;
-                const fileName = `export_${timeStamp}.${extension}`;
-                saveAs(resp, fileName);
-            },
-            (error: APIError) => console.log('error.msg', error.msg)
-        );
+        const exportObservable: Observable<object> = this.apiService.exportRecords(this.dateStart, this.dateEnd,
+            this.speciesName, this.selectedDataset.id, this.fileType);
+
+        const saveBlob = function(blob: Blob) {
+            const timeStamp = moment().format('YYYY-MM-DD-HHmmss');
+            const extension = this.fileType;
+            const fileName = `export_${timeStamp}.${extension}`;
+            saveAs(blob, fileName);
+        }.bind(this);
+
+        if (this.lockRecordsOnExport) {
+            const lockingObservalble = this.apiService.getRecordsByDatasetId(this.selectedDataset.id,
+                    JSON.parse(JSON.stringify(this.recordParams))).pipe(
+                mergeMap((records: Record[]) => from(records).pipe(
+                    mergeMap((record: Record) => this.apiService.updateRecordLocked(record.id, true))
+                ))
+            );
+
+            this.isLocking = true;
+
+            forkJoin(exportObservable, lockingObservalble).subscribe(
+                (results: [Blob, any]) => saveBlob(results[0]),
+                (error: APIError) => {
+                    this.messageService.add({
+                        severity: 'error',
+                        summary: 'Export error',
+                        detail: `There were error(s) when exporting / locking: ${error.msg}`
+                    });
+                    this.isLocking = false;
+                },
+                () => {
+                    if (this.lockRecordsOnExport) {
+                        this.messageService.add({
+                            severity: 'success',
+                            summary: 'Export success',
+                            detail: `There records were successfully exported and have now been locked`
+                        });
```

Comment:
Don't we also want a message even when  `this.lockRecordsOnExport ` is `false` ? Maybe append `and have now been locked` in the message if `this.lockRecordsOnExport`

Reply:
Oh I didn't see we were in a outer `if (this.lockRecordsOnExport)`. So maybe the second `if (this.lockRecordsOnExport)` is not necessary.

Reply:
Yes, I'll remove this inner if

Reply:
Removed